theano/gpuarray/opt.py

```diff
@@ -210,6 +210,52 @@ def apply(self, fgraph):
                     0, 'fast_run', 'fast_compile', 'merge')
 
 
+class GraphToGPU(Optimizer):
+    """
+    Transfer the graph as a whole to GPU instead of replacing nodes
+    """
+    def add_requirements(self, fgraph):
+        fgraph.attach_feature(toolbox.ReplaceValidate())
+
+    def apply(self, fgraph):
+        mapping = {}
+        move_to_GPU = True
+
+        # Building a new graph
+        for i in fgraph.inputs:
+            mapping[i] = GpuFromHost(None)(i)
```

Comment:
> Here in the current code, we don't iter on the nodes on the graph, but only on the inputs of the graph.
The dic map a Theano variable in the old graph to the corresponding Theano variable in the new graph. I think the current code it good.

Yeah, I understand this. But, if i initialise the dict with the Variable that are input to the graph, and later iterate this dictionary with Variable that are input to an Apply node, would there be some extra Variable in the latter ?